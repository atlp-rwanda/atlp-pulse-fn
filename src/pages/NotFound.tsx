import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

function NotFound() {
  const { t } = useTranslation();
  return (
    <div className="notFound dark:bg-dark-frame-bg flex flex-row py-12 px-5  md:flex-row items-center justify-end md:justify-evenly grow h-full w-full ">
      <div />
      <div className="bg-light-bg dark:bg-dark-bg px-4 py-5 rounded-3xl max-w-md flex flex-col h-full">
        <h3 className="text-2xl font-sans font-bold dark:text-dark-text-fill ">
          {t('Ooops... What just happened!?')}
        </h3>
        <h2 className="text-[4rem] font-bold text-primary ">{t('NOT FOUND')}</h2>
        <p className="text-base font-bold dark:text-dark-text-fill whitespace-pre-wrap">
          {t(
            '404_page_paragraph',
          )}
        </p>
        <Link
          className="px-5 py-3 bg-primary font-bold mt-5 rounded-md text-white w-fit"
          to="/"
        >
          {t('Back Home')}
        </Link>
      </div>
    </div>
  );
}

export default NotFound;
