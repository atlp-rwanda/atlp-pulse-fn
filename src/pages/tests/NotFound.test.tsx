import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import NotFound from '../NotFound';

describe('Not Found', () => {
  it('Renders the 404 Page', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <NotFound />
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
