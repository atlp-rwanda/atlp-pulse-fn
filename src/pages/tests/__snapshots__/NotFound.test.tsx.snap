// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Renders the 404 Page 1`] = `
<div
  className="notFound dark:bg-dark-frame-bg flex flex-row py-8 px-5  md:flex-row items-center justify-end md:justify-evenly grow h-full w-full "
>
  <div />
  <div
    className="bg-light-bg dark:bg-dark-bg px-4 py-5 rounded-3xl max-w-md flex flex-col h-full"
  >
    <h3
      className="text-2xl font-sans font-bold dark:text-dark-text-fill "
    >
      Ooops... What just happened!?
    </h3>
    <h2
      className="text-[4rem] font-bold text-primary "
    >
      NOT FOUND
    </h2>
    <p
      className="text-base font-bold dark:text-dark-text-fill whitespace-pre-wrap"
    >
      You did not break the internet ðŸ˜±. The page you are looking for does not exists.
    </p>
    <a
      className="px-5 py-3 bg-primary font-bold mt-5 rounded-md text-white w-fit"
      href="/"
      onClick={[Function]}
    >
      Back Home
    </a>
  </div>
</div>
`;
